### PR TITLE
Tron, Ontology, ZCash support (base58check)

### PR DESCRIPTION
**Pull request by catenocrypt**

### CODE REVIEW

File: zcash.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+func ZCashDecodeToBytes(input string) ([]byte, error) {
+	decoded, err := Base58ChecksumDecode(input, Base58DefaultAlphabet)
+	if err != nil {
+		return nil, err
+	}
+	if len(decoded) != ZCashAddressLength {
+		return nil, errors.New("Invalid length")
+	}
+	// prefix check
+	staticPrefix := decoded[0]
```
Comment (hewigovens) [inline zcash.go]: Zcash has two bytes prefix, encoded as t1 or t3

File: zcash_test.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+		{
+			name:   "Normal",
+			input:  "t1RygJmrLdNGgi98gUgEJDTVaELTAYWoMBy",
+			output: "1cb858e71790e51ab7558c05a6067cfc4926aa8c44dd",
+		},
+		{
+			name:   "Normal2",
+			input:  "t1TWk2mmvESDnE4dmCfT7MQ97ij6ZqLpNVU",
+			output: "1cb869bf38acef973293c07f05c778eb1209748e8d52",
+		},
```
Comment (hewigovens) [inline zcash_test.go]: Add one more t3 address test

File: zcash.go (hunk 1 above)
Comment (catenocrypt) [reply zcash.go]: I based on walletcore implementation, and there is no prefix check there at all.  It accepts this as valid: "t488N47kTBknqPow5zKQbZjKTGKRpJkh4ab".  Should I check here to accept t1 and t3 only?  We don't need to be very strict here, important is that ENS produces the same input.

File: zcash_test.go (hunk 2 above)
Comment (catenocrypt) [reply zcash_test.go]: Added

File: zcash.go (hunk 1 above)
Comment (hewigovens) [reply zcash.go]: hmm, then looks like it's a wallet core bug

**Pull request merged**